packages/csharp/PortingAssistant/PortingAssistant.UnitTests/PortingAssistantUtilsTests.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using NUnit.Framework;
+using PortingAssistant.Common.Utils;
+
+
+namespace PortingAssistant.UnitTests
+{
+
+    public class PortingAssistantUtilsTests
+    {
+        private static Random random = new Random();
+        private const int MaxPathLength = 260;
+
+
+        [Test]
+        public void TestSolutionPathTooLongWillThrowException()
+        {
+            var solutionPath = RandomString(265);
+            var destinationPath = "xx/yy.sln";
+            try
+            {
+                PortingAssistantUtils.CopyDirectory(solutionPath, destinationPath);
+            }
+            catch (Exception e)
+            {
+                Assert.AreEqual(e.Message,
+                    $"The solution path length cannot exceed {MaxPathLength} characters. Please try a location that has a shorter path.");
+            }
+        }
+
+        [Test]
+        public void TestFileNamePlusDestinationPathTooLongWillThrowException()
+        {
+            var solutionPath = "xx/yy.sln";
+            var destinationPath = RandomString(258);
+
+            try
+            {
+                PortingAssistantUtils.CopyDirectory(solutionPath, destinationPath);
+            }
+            catch (Exception e)
+            {
+                Assert.AreEqual(e.Message,
+                    $"The destination path length cannot exceed {MaxPathLength} characters. Please try a location that has a shorter path.");
+            }
+        }
```

Comment:
Agree with Sai, good work with the test cases. I do have two more suggestions:
1. Instead of placing the `Assert` statement in the `catch` block, NUnit (and other testing frameworks) have a preferred way of asserting exceptions:
```csharp
Assert.Throws<SomeException>(() => 
{ 
    SomeMethodThatThrowsAnException(params);
});
```

2. Similarly, we can inverse those kinds of assertions:
```csharp
Assert.Throws<SomeException>(() => 
{ 
    SomeMethodThatDoesNotThrowAnException(params);
});
```
I would recommend adding a test case for when solutionPath + destinationPath == 260 chars (it should not throw an error, but I suspect it will based on logic in line 40 of `PortingAssistantUtils.cs`)

Reply:
Addressed comments.